src/js/models/layerstructure.js

```diff
@@ -1328,8 +1328,80 @@ define(function (require, exports, module) {
     };
 
     /**
+     * Set basic properties of the layerEffect at the given index of the given layers.
+     *
+     * @param {Immutable.Iterable.<number>} layerIDs
+     * @param {number | Immutable.List.<number>} layerEffectIndex index of effect, or per-layer List thereof
+     * @param {string} layerEffectType type of layer effect
+     * @param {object | Immutable.List.<object>} layerEffectProperties properties to merge, or per-layer List thereof
+     * @return {LayerStructure}
+     */
+    LayerStructure.prototype.setLayerEffectProperties = function (layerIDs,
+        layerEffectIndex, layerEffectType, layerEffectProperties) {
+        // validate layerEffectType
+        if (!Layer.layerEffectTypes.has(layerEffectType)) {
+            throw new Error("Invalid layerEffectType supplied");
+        }
+
+        var nextLayers = Immutable.Map(layerIDs.reduce(function (map, layerID, index) {
+            var layer = this.byID(layerID),
+                layerEffects = layer.getLayerEffectsByType(layerEffectType) || Immutable.List(),
+                _layerEffectIndex,
+                layerEffect,
+                nextLayerEffect,
+                newProps,
+                nextLayer;
+
+            _layerEffectIndex = Immutable.List.isList(layerEffectIndex) ?
```

Comment:
These are already existed in the master, don't know why it is showing as new here. Do you want me to fix them here?